pkg/resource/deploy/step_generator.go

```diff
@@ -1034,7 +1034,9 @@ func (sg *stepGenerator) generateStepsFromDiff(
 					"program and run `pulumi up`", urn)
 				sg.deployment.ctx.Diag.Errorf(diag.StreamMessage(urn, message, 0))
 				sg.sawError = true
-				return nil, result.BailErrorf(message)
+				if !sg.deployment.preview {
```

Comment:
Probably worth a comment why we have to error here if not preview.

Reply:
I added a comment. I'm not exactly sure we _need_ to bail during update. Honestly, I'm making the smallest possible change to solve the preview issue.

Reply:
If we don't bail and we're doing a create before delete replacement we'll execute the create before getting to the delete error. State will still be tracking both resources but it's an awkward state to cleanup from.